internal/app/machined/pkg/controllers/network/route_config.go

```diff
@@ -0,0 +1,297 @@
+	ids := make([]string, 0, len(routes))
+
+	for _, route := range routes {
+		route := route
+		id := network.LayeredID(route.Layer, network.RouteID(route.Destination, route.Gateway))
+
+		if err := r.Modify(
+			ctx,
+			network.NewRouteSpec(network.ConfigNamespaceName, id),
+			func(r resource.Resource) error {
+				*r.(*network.RouteSpec).Status() = route
+
+				return nil
+			},
+		); err != nil {
+			return ids, err
+		}
+
+		ids = append(ids, id)
+	}
+
+	return ids, nil
+}
+
+func (ctrl *RouteConfigController) parseCmdline(logger *zap.Logger) (route network.RouteSpecSpec) {
+	if ctrl.Cmdline == nil {
+		return
+	}
+
+	settings, err := ParseCmdlineNetwork(ctrl.Cmdline)
+	if err != nil {
+		logger.Info("ignoring error", zap.Error(err))
+
+		return
+	}
+
+	if settings.Gateway.IsZero() {
+		return
+	}
+
+	route.Gateway = settings.Gateway
+
+	if route.Gateway.Is6() {
+		route.Family = nethelpers.FamilyInet6
+	} else {
+		route.Family = nethelpers.FamilyInet4
+	}
+
+	route.Scope = nethelpers.ScopeGlobal
+	route.Table = nethelpers.TableMain
+	route.Priority = DefaultRouteMetric
+	route.Protocol = nethelpers.ProtocolBoot
+	route.Type = nethelpers.TypeUnicast
+	route.OutLinkName = settings.LinkName
+	route.Layer = network.ConfigCmdline
+
+	return route
+}
+
+var (
+	zero16 = netaddr.MustParseIP("::")
+	zero4  = netaddr.MustParseIP("0.0.0.0")
+)
+
+//nolint:gocyclo
+func (ctrl *RouteConfigController) parseMachineConfiguration(logger *zap.Logger, cfgProvider talosconfig.Provider) (routes []network.RouteSpecSpec) {
+	convert := func(linkName string, in talosconfig.Route) (route network.RouteSpecSpec, err error) {
+		if in.Network() != "" {
+			route.Destination, err = netaddr.ParseIPPrefix(in.Network())
+			if err != nil {
+				return route, fmt.Errorf("error parsing route network: %w", err)
+			}
+
+			if route.Destination.Bits == 0 && (route.Destination.IP.Compare(zero4) == 0 || route.Destination.IP.Compare(zero16) == 0) {
+				// clear destination to be zero value to support "0.0.0.0/0" routes
+				route.Destination = netaddr.IPPrefix{}
+			}
+		}
+
+		route.Gateway, err = netaddr.ParseIP(in.Gateway())
+		if err != nil {
+			return route, fmt.Errorf("error parsing route gateway: %w", err)
+		}
+
+		route.Priority = in.Metric()
+		if route.Priority == 0 {
+			route.Priority = DefaultRouteMetric
+		}
+
+		if route.Gateway.Is6() {
+			route.Family = nethelpers.FamilyInet6
+		} else {
+			route.Family = nethelpers.FamilyInet4
+		}
+
+		route.Table = nethelpers.TableMain
+		route.Protocol = nethelpers.ProtocolStatic
+		route.OutLinkName = linkName
+		route.Layer = network.ConfigMachineConfiguration
+
+		route.Scope = nethelpers.ScopeGlobal
+
+		if route.Destination.IP.IsLinkLocalUnicast() || route.Destination.IP.IsLinkLocalMulticast() {
+			route.Scope = nethelpers.ScopeLink
+		} else if route.Destination.IP.IsLoopback() {
```

Comment:
In general, I think `switch` is often more clear than `else if`:

```go
switch {
case route.Destination.IP.IsLinkLocalUnicast():
  fallthrough
case route.Destination.IP.IsLinkLocalMulticast():
  route.Scope = nethelpers.ScopeLink
case route.Destination.IP.IsLoopback():
  route.Scope = nethelpers.ScopeHost
}
```

Reply:
+1, not a blocker for me.

Reply:
I don't like `fallthrough` unless it's really needed, so this boils down just to two clauses (first one with `||`), I don't know if it's worth it here? I fully agree that 3 and more are better as `switch`

Reply:
I don't have a strong opinion either way

Reply:
I ended up refactoring it to use switch, with `default:` case it goes to 3 clauses!